hbase-common/src/main/java/org/apache/hadoop/hbase/io/ByteBufferListOutputStream.java

```diff
@@ -39,37 +41,28 @@
 public class ByteBufferListOutputStream extends ByteBufferOutputStream {
   private static final Logger LOG = LoggerFactory.getLogger(ByteBufferListOutputStream.class);
 
-  private ByteBufferPool pool;
+  private ByteBuffAllocator allocator;
   // Keep track of the BBs where bytes written to. We will first try to get a BB from the pool. If
   // it is not available will make a new one our own and keep writing to that. We keep track of all
   // the BBs that we got from pool, separately so that on closeAndPutbackBuffers, we can make sure
   // to return back all of them to pool
-  protected List<ByteBuffer> allBufs = new ArrayList<>();
-  protected List<ByteBuffer> bufsFromPool = new ArrayList<>();
+  protected List<SingleByteBuff> allBufs = new ArrayList<>();
```

Comment:
Will a MultiByteBuff be better here?

Reply:
Consider this before, seems can not because the buffer list in ByteBufferListOutputStream can extend dynamiclly,  because we may write an unknown byte size data.  while the ByteBuffer list in MultiByteBuff is mutable now.